bed/bedPe/bedPe.go

```diff
@@ -0,0 +1,185 @@
+// Package bedPe provides functions for handling Bed Paired End format files, used for
+// handling pairs of genomic regions.
+
+package bedPe
+
+import (
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"io"
+	"log"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type BedPe struct {
+	Chrom1            string
+	ChromStart1       int
+	ChromEnd1         int
+	Chrom2            string
+	ChromStart2       int
+	ChromEnd2         int
+	Name              string
+	Score             int
+	Strand1           bed.Strand
+	Strand2           bed.Strand
+	FieldsInitialized int      //number of fields that are initialized, used for smart writing.
+	Annotation        []string //long form for extra fields
```

Comment:
Do we need this?  I think we have this in bed as a bit of a hack to make things work, but it might be good to not propagate it if we can.

Reply:
it's messy, but this is the bedpe standard, additional tab delimited fields at the user's discretion, just like bed.

Reply:
I would be in favor of replacing the []string in bed and bedPe with an unparsed tab-delimited string. A string here would have a few advantages
1) less heap memory allocation
2) you can do whole-struct comparisons (bedA == bedB)
3) you can use beds as keys in maps (something that I have recently been doing by converting beds to a custom struct that removes the annotation field)

Reply:
I like Dan's idea here.  I think we should also think about applying it to "bed" (although that would be a separate PR) sometime in the future.  Do we have any code/use-case where keeping it unparsed would be really bad?

Reply:
Are you going to change the bedpe struct and the Annotation []string in this commit? If not, then I am ready to approve when the names of folders and functions are all fixed.